infrastructure/testdriver/bidi/speculation/prefetch_status_updated.https.html

```diff
@@ -0,0 +1,181 @@
+<!DOCTYPE html>
+<meta charset="utf-8"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/dispatcher/dispatcher.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/resources/testdriver.js?feature=bidi"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="resources/bidi-speculation-helper.js"></script>
+
+<script>
+    promise_setup(async () => {
+        await waitForDocumentReady();
+        await test_driver.bidi.speculation.prefetch_status_updated.subscribe();
+    });
+
+    promise_test(async t => {
+
+        const receivedEvents = [];
+        const expectedEvents = ['pending', 'ready'];
+
+        // Create a promise that resolves when we receive the 'ready' event
+        
+        const readyEventPromise = new Promise((resolve, reject) => {
+            const removeHandler = test_driver.bidi.speculation.prefetch_status_updated.on((event) => {
+                receivedEvents.push(event);
+                
+                // Status should be one of the expected values
+                const validStatuses = ['pending', 'running', 'ready', 'success', 'failure'];
+                assert_true(validStatuses.includes(event.status), 
+                           `Event status should be one of ${validStatuses.join(', ')}, got: ${event.status}`);

```

Comment:
I don't this assertion is needed, as it is restrictive for other potential extensions of the speculation rules BiDi API.